processing/src/main/java/io/druid/segment/IndexMerger.java

```diff
@@ -470,20 +480,71 @@ public String apply(@Nullable String input)
       }
     };
 
-    return makeIndexFiles(indexes, outDir, progress, mergedDimensions, mergedMetrics, null, rowMergerFn, indexSpec);
+    return makeIndexFiles(
+        indexes,
+        mergeAggregators(indexes),
+        outDir,
+        progress,
+        mergedDimensions,
+        mergedMetrics,
+        null,
+        rowMergerFn,
+        indexSpec
+    );
+  }
+
+  public static final AggregatorFactory[] mergeAggregators(List<IndexableAdapter> indexes)
+  {
+    Set<AggregatorFactory> mergedAggregators = Sets.newTreeSet(
+        Ordering.from(
+            new Comparator<AggregatorFactory>()
+            {
+              @Override
+              public int compare(AggregatorFactory a1, AggregatorFactory a2)
+              {
+                String name1 = a1.getName();
+                String name2 = a2.getName();
+                if (name1 == null) {
+                  return -1;
+                } else {
+                  return name1.compareTo(name2);
+                }
+              }
+            }
+        ).nullsFirst()
+    );
+    for (IndexableAdapter index : indexes) {
```

Comment:
I feel like this is a bit too loose. It's basically saying that if there are two aggregators with the same name, just pick one somehow. It's totally possible for two aggregators with the same name to actually be different.

Maybe we should only include metadata for aggregators that are actually equal. We could leave the others out and leave it up to the user to decide if they want to query them or not. Failing the merge is also an option, although if we do that then I think it's important to make sure that people who change aggregator types, for whatever reason, aren't screwed by coordinator merge/reindex tasks continuously failing and things getting stuck.